store/store_test.go

```diff
@@ -481,10 +409,6 @@ func TestPruneBlocks(t *testing.T) {
 		require.NotNil(t, bs.LoadBlock(i))
 	}
 
-	// Pruning below the current base should error
-	_, err = bs.PruneBlocks(1199)
```

Comment:
I kind of disagree here. I think it's okay not to return an error if the height is less than the current base. Semantically it's the same and the pruning function has done it's job - there are no blocks below this height. 

I may be wrong though so speak up if anyone disagrees

Reply:
Yeah, I'm inclined to agree. I think the idea here was that if you pruned to height 1000 then you would expect height 1000 to exist afterwards. If 1000 is below the base, then it won't exist after pruning to it. Not sure if that makes sense or not, don't have a strong intuition for what's the better behavior.

Reply:
Yeah I guess you've just somewhat inverted my assumption into: what blocks should exist after pruning? in which it might seem more intuitive to return an error for something below the base. 

Currently the state store also doesn't return an error if it is below the earliest recorded height. 

In the only use of pruning (in `finalizeCommit()` by state) there is already a check to make sure that the retain height is not below the base height

Reply:
Good point, the caller can check this for themselves if they need to.